build_pipelines/aspnetcorebuild-1.0/prepare_project.py

```diff
@@ -23,23 +23,32 @@
 
 """
 
+import argparse
 import glob
 import os
 import sys
 import textwrap
 
 
+# Arguments for the builder.
+PARSER = argparse.ArgumentParser()
+PARSER.add_argument('-r', '--runtime-image',
+                    dest='runtime_image',
+                    help='The runtime image to use for the Dockerfile.',
+                    required=True)
+
+
 ASSEMBLY_NAME_TEMPLATE = '{0}.dll'
 DEPS_PATTERN = '*.deps.json'
 DEPS_EXTENSION = '.deps.json'
 DOCKERFILE_NAME = 'Dockerfile'
 DOCKERFILE_CONTENTS = textwrap.dedent(
     """\
-    FROM gcr.io/google-appengine/aspnetcore@sha256:a5cb3f4a9be727ed449771d8018f29696a53bf9116a94b81c3d7719cb97b99af
+    FROM {0}
```

Comment:
nit: For a format string this long, using named parameters might be clearer:

   FROM {runtime_image}
   ...
   ENTRYPOINT [  "dotnet", "{dll_name}.dll"
   """.format(runtime_image='foo', dll_name='bar')